ShowerShow/Control/UserPreferencesController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+using ShowerShow.Controllers;
+using ShowerShow.DTO;
+using ShowerShow.Models;
+using ShowerShow.Repository.Interface;
+using ShowerShow.Service;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ShowerShow.Control
+{
+    public class UserPreferencesController
+    {
+        private readonly ILogger<UserPreferencesController> _logger;
+        private IUserPrefencesService _userPrefencesService;
+        private IUserService _userService;
+        public UserPreferencesController(ILogger<UserPreferencesController> logger, IUserPrefencesService userPrefencesService, IUserService userService)
+        {
+            _logger = logger;
+            this._userPrefencesService = userPrefencesService;
+            this._userService = userService;
+
+        }
+
+        [Function("CreateUserPreferences")]
+        [OpenApiOperation(operationId: "CreatePreferences", tags: new[] { "Preferences" })]
+        [OpenApiRequestBody("application/json", typeof(CreatePreferencesDTO), Description = "Id of the requested user ")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.Created, contentType: "application/json", bodyType: typeof(CreatePreferencesDTO), Description = "Successfully created the Preferences")]
+        public async Task <HttpResponseData> CreateUserPreferences([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "user/{userId}/preferences")] HttpRequestData req)
+        {
+            _logger.LogInformation("Create a preference for one user");
+
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+
+            CreatePreferencesDTO createPreferencesDTO = JsonConvert.DeserializeObject<CreatePreferencesDTO>(requestBody);
+
+            if (await _userService.CheckIfUserExistAndActive(createPreferencesDTO.UserId))
```

Comment:
For this case its correct you should only create pref if it doesnt exist this doesnt need to be fixed i think